src/client/core/utils/content-editable.js

```diff
@@ -368,6 +378,35 @@ export function getSelectionEndPosition (el, selection, inverseSelection) {
     return calculatePositionByNodeAndOffset(el, correctedSelectionEnd);
 }
 
+function getElementOffset (target) {
+    let offset = 0;
+
+    const firstBreakElement = arrayUtils.find(target.childNodes, (node, index) => {
+        offset = index;
+        return domUtils.getTagName(node) === 'br';
+    });
+
+    return firstBreakElement ? offset : 0;
+}
+
+function isNodeSelectable (node, includeDescendants = false) {
+    if (domUtils.isTextNode(node))
+        return true;
+
+    if (!domUtils.isElementNode(node))
+        return false;
+
+    const visibleChildren       = getVisibleChildren(node);
+    const isContentEditableRoot = !domUtils.isContentEditableElement(node.parentNode);
+    const selectableChildren    = arrayUtils.filter(visibleChildren, child => isNodeSelectable(child));
+    const hasBreakLineElements  = arrayUtils.some(visibleChildren, child => domUtils.getTagName(child) === 'br');
+
+    if (selectableChildren.length)
```

Comment:
Let's add `includeDescendants`  arg to `hasSelectableChildren` method and use it here